cfme/tests/cloud/test_providers.py

```diff
@@ -538,6 +540,39 @@ def test_azure_multiple_subscription(appliance, request, soft_assert):
                                                                                      prov_b[0]))
 
 
+@pytest.mark.tier(3)
+@test_requirements.azure
+@pytest.mark.meta(automates=[1495318], blockers=[BZ(1756984)])
+@pytest.mark.provider([AzureProvider], scope="function", override=True, selector=ONE)
+def test_refresh_with_empty_iot_hub_azure(request, provider, setup_provider):
+    """
+    Polarion:
+        assignee: anikifor
+        casecomponent: Cloud
+        caseimportance: low
+        initialEstimate: 1/6h
+        setup: prepare env
+               create an IoT Hub in Azure (using free tier pricing is good enough):
+               $ az iot hub create --name rmanes-iothub --resource-group iot_rg
+        testSteps:
+            1. refresh azure provider
+        expectedResults:
+            1. no errors found in logs
+    Bugzilla:
+        1495318
+    """
+    result = LogValidator("/var/www/miq/vmdb/log/evm.log", failure_patterns=[r".*ERROR.*"])
+    result.start_monitoring()
+    azure = provider.mgmt
+    if not azure.has_iothub():
+        azure.create_iothub("potatoiothub")
```

Comment:
This needs to be randomized. If we're running multiple instances of the test framework concurrently, they could collide.

Reply:
just a note: they could collide even in that case too. In our azure we can't have more than 1 iothub (1 is free).